File: 프로그래머스/lv1/92334. 신고 결과 받기/신고 결과 받기.py
def solution(id_list, report, k):
    report_cnt = {i: 0 for i in id_list}
    answer = [0] * len(id_list)
    id_list = {id_list[i]: i for i in range(len(id_list))}
    report = set(report)
    
    for r in report:
        reporter, reported = r.split(" ")
        report_cnt[reported] += 1
        
    for r in report:
        reporter, reported = r.split(" ")
        if report_cnt[reported] >= k:
            answer[id_list[reporter]] += 1
        
    
    return answer

File: 프로그래머스/1/92334. 신고 결과 받기/신고 결과 받기.js
function solution(id_list, report, k) {
    const answer = new Array(id_list.length).fill(0)
    const reportCnt = new Array(id_list.length).fill(0)
    const idMap = id_list.reduce((acc, cur, idx) => {
        acc[cur] = idx
        return acc
    }, {})
    
    const reportSet = new Set(report)
    
    for (let i of reportSet) {
        const reported = i.split(" ")[1]
        reportCnt[idMap[reported]]++
    }
    
    for (let i of reportSet) {
        const [reporter, reported] = i.split(" ")
        if (reportCnt[idMap[reported]] >= k) {
            answer[idMap[reporter]]++
        }
    }
    return answer;
}